Add optimized nearestImage method to Vector

akofke opened the pull request:
Add a specialized nearestImage method for BoundaryRectangularPeriodic to Vector.

akofke committed:
Add optimized nearestImage method to Vector

Have BoundaryRectangularPeriodic call this method for nearestImage.

akofke commented:
@ajschult this causes the snapshot test to fail so I'm not sure it's entirely correct or at least does things differently than the other implementation. Maybe from rounding differences due to fewer floating point operations?

ajschult commented:
Yes, I would be surprised if this sort of change didn't cause the snapshot to fail.  The snapshot test is just asserting that you haven't meaningfully changed anything related to floating point computation (which is pretty much everything).

... do you need dimensionsHalf?

akofke committed:
Change Vector nearestImage to only need dimensions

Add nearestImage benchmark. Benchmark shows there is no difference
between passing dimensionsHalf or calculating them in the method.

akofke committed:
BoundaryRectangularPeriodic no longer needs dimensionsHalf

akofke committed:
Add tests for BoundaryRectangularPeriodic

Add a static standard delta field to UnitTestUtil.

akofke committed:
Update snapshot test coordinates file

The changes in coordinates are due to floating point rounding differences,
so we can assume they're still correct.
